lib/cocoapods-downloader/git.rb

```diff
@@ -23,9 +23,15 @@ def checkout_options
       def self.preprocess_options(options)
         return options unless options[:branch]
 
+        input = [options[:git], options[:commit]].map(&:to_s)
+        invalid = input.compact.any? { |value| value.start_with?('--') || value.include?(' --') }
+        raise DownloaderError, "Provided unsafe input for git #{options}." if invalid
+
         command = ['ls-remote',
+                   '--',    
```

Comment:
`--` was added to the version of git shipped with 2.25, but version 2.24 is still pretty actively used by default in osx as Apple Git-128 somewhere about 2 years ago - we shouldn't rely solely on the `--` IMO

I'm trying to get an old git to test that this param is ignored ATM